src/ripple/ledger/tests/View_test.cpp

```diff
@@ -524,6 +525,209 @@ class View_test
     }
 
     void
+    testFlags()
+    {
+        using namespace jtx;
+        Env env(*this);
+
+        auto const alice = Account("alice");
+        auto const bob = Account("bob");
+        auto const carol = Account("carol");
+        auto const gw = Account("gateway");
+        auto const USD = gw["USD"];
+        auto const EUR = gw["EUR"];
+
+        env.fund(XRP(10000), alice, bob, carol, gw);
+        env.trust(USD(100), alice, bob, carol);
+        {
+            // Global freezing.
+            env(pay(gw, alice, USD(50)));
+            env(offer(alice, XRP(5), USD(5)));
+
+            // Now freeze gw.
+            env(fset (gw, asfGlobalFreeze));
+            env.close();
+            env(offer(alice, XRP(4), USD(5)), ter(tecFROZEN));
+            env.close();
+
+            // Alice's USD balance should be zero if frozen.
+            expect(USD(0) == accountHolds (*env.closed(),
+                alice, USD.currency, gw, fhZERO_IF_FROZEN, env.journal));
+
```

Comment:
The purpose of this test is to exercise functions in `View.cpp` to get coverage up to 100%, not to verify the correct behavior of any particular business logic (except that necessary to get coverage up).

That's not to say that we should not have such tests for freeze, but they belong in a unit test that specifically addresses the verification of correct freeze functionality.